internal/provider/datasource_port_availability.go

```diff
@@ -0,0 +1,98 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/PacketFabric/terraform-provider-packetfabric/internal/packetfabric"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourcePortAvailability() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePortAvailabilityRead,
+		Schema: map[string]*schema.Schema{
+			"pop": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "The port POP.",
+			},
+			"ports_available": {
+				Type:     schema.TypeList,
+				Computed: true,
```

Comment:
@pontovinte @elXaptation could we add a description for locations_zones?

Something like Description: `"The list of ports available in the given POP.",`